federation/pkg/kubefed/init/init.go

```diff
@@ -173,6 +176,7 @@ func (o *initFederationOptions) Bind(flags *pflag.FlagSet, defaultServerImage, d
 	flags.Int32Var(&o.apiServerNodePortPort, apiserverPortFlag, 0, "Preferred port to use for api server nodeport service (0 for random port assignment). Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
 	flags.BoolVar(&o.apiServerEnableHTTPBasicAuth, "apiserver-enable-basic-auth", false, "Enables HTTP Basic authentication for the federation-apiserver. Defaults to false.")
 	flags.BoolVar(&o.apiServerEnableTokenAuth, "apiserver-enable-token-auth", false, "Enables token authentication for the federation-apiserver. Defaults to false.")
+	flags.StringVar(&o.tolerationsString, "tolerations", "", "If specified, the pod's tolerations. Example \"key1=value1:NoSchedule,key2=value2:NoSchedule\"")
```

Comment:
@tengqm I'm divided on this.  On the one hand we could have the user/administrator able to customize the deployment template to do whatever they want (toleration in this case).  On the other hand the intention of this kube-fed tool is to build the deployment definition on the basis of user-provided flags (with sensible default values).  One approach might be to expose the template in such a way that it's easily customizable for power users who need to do things that the toll does not currently support, but try to have the tool support as many sensible command line flags as is feasible, so that the user does not have to craft custom templates.  I'll leave it to @madhusudancs to decide how he wishes to proceed, as he's given much more thought to this than I have.  I would be inclined to support both (in this case both the tolerations flag, as well as template customization).